engine/pkg/clients/internal/call.go

```diff
@@ -0,0 +1,38 @@
+// Copyright 2022 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package internal
+
+import (
+	"context"
+
+	"google.golang.org/grpc"
+
+	"github.com/pingcap/tiflow/engine/pkg/rpcerror"
+)
+
+// Call calls a grpc client function.
+func Call[ReqT any, RespT any, F func(context.Context, ReqT, ...grpc.CallOption) (RespT, error)](
```

Comment:
Is it compatible with `grpc.UnaryClientInterceptor`?

Reply:
I have a feeling that using an interceptor is not good for code maintainability. But it seems for now using a unary interceptor is fine.

Reply:
The `interceptor` is like middleware in http. It is used in many places, like prometheus, etcd and other retry utils. It can help us simplify the code and improve code maintainability.